src/main/java/com/fasterxml/jackson/databind/introspect/EnumPropertiesCollector.java

```diff
@@ -0,0 +1,104 @@
+package com.fasterxml.jackson.databind.introspect;
+
+import com.fasterxml.jackson.databind.AnnotationIntrospector;
+import com.fasterxml.jackson.databind.EnumNamingStrategy;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.annotation.EnumNaming;
+import com.fasterxml.jackson.databind.cfg.MapperConfig;
+import com.fasterxml.jackson.databind.util.ClassUtil;
+
+/**
+ * Helper class used for aggregating information about all possible
+ * properties of a Enum.
+ *
+ * @since 2.15
+ */
+public class EnumPropertiesCollector {
```

Comment:
I think this class can be replaced (at least mostly) by moving construction of Enum lookup names, strategy to an earlier part when constructing Enum[De]Serializer -- at the point where we have `AnnotatedClass` available for Enum type.
Lookup functionality could perhaps be added in `MapperConfigBase` (base type of `DeserializationConfig`/`SerializationConfig`), being passed `AnnotatedClass` of enum type.